docs/proposals/secure-impersonation.md

```diff
@@ -0,0 +1,636 @@
+cluster-admin ClusterRoleBinding.
+This means cluster owners need to be careful who can create Kustomizations and HelmReleases as well as which Sources are in the cluster
+due to the exposure they provide to act as cluster-admin.
+This restricts who can create Source and Apply type objects into a burdensome approval flow for cluster owners and provides
+poor delegation of self-service permissions which is a proven elements of good platforms.
+
+
+## Goals
+
+Flux should remain easy to install but promote secure practices in its default configuration.
+Flux's security model should allow a cluster administrator to delegate self-service management of
+Sources, Applies, Webhooks, Automations, and Notifications to Namespace owners/users without having to review every detail for
+compliance to security policies.
+Tenants or Namespace owners should be able to act freely without fear that their changes will overreach their permissions.
+
+Flux APIs must structurally prevent privilege escalation and arbitrary code execution.
+Flux APIs should be congruent with Kubernetes core API's security model.
+
+The implementation should be usable and allow users quick, concise decisions regarding their security needs.
+
+
+## Design
+
+Currently kustomize-controller and helm-controller apply Kustomizations and HelmReleases using their own ServiceAccount by default.
+It's possible to drop privileges using a `serviceAccountName` from the same Namespace -- this is more secure but this is an optional, explicit behavior.
+
+The central controllers should change so they always assume a different User when applying and maintaining objects from manifests in Sources.
+They should never use a controller SA when managing these resulting Objects, even for Garbage Collection, Health-Checks, or "values-from" merging.
+This will allow the audit log to be more granularly examined for controllers' mutations on Flux APIs vs. GitOps-managed objects.
+This technique prevents cross-tenant information disclosure and privilege escalation through misused Health-Checks or Garbage Collection.
+
+#### controller serviceAccounts
+The controller ServiceAccounts are far overprivileged for Flux API operations.
+They should have a ClusterRoleBinding (or RoleBinding for single Namespace installs) for necessary Flux API Kinds and other needed objects such as ConfigMaps/Secrets + a separate ClusterRoleBinding for User Impersonation.
+
+#### user/group names
+All usernames impersonated by kustomize-controller and helm-controller for managing resources will be Namespaced relative to the `Kustomization`/`HelmRelease` object.
+This is similar to the mechanism Kubernetes uses for [ServiceAccount User names](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-subjects)).
+Kubernetes uses `system:serviceaccount:{{namespace}}:{{metadata.name}}` for ServiceAccounts.
+The Flux username format should be `flux:user:{{namespace}}:{{spec.user}}`.
```

Comment:
What use cases do we see for making the name configurable? I can't see anything other than making it possible to lower the impact of something by decision of the creator.

If tenant-A grants tenant-B access (edit role) to the whole namespace tenant-A for reconciler but then grants another role (pod-deleter) to the user pod-deleter, then it would be up to tenant-B to know and decide on which one to use.

If there isn't a strong enough case for these scenarios - maybe the user spec isn't needed and a default one is always used?